cineast-core/src/main/java/org/vitrivr/cineast/core/db/cottontaildb/CottontailWrapper.java

```diff
@@ -1,339 +1,54 @@
-    final CottonDDLBlockingStub stub = CottonDDLGrpc.newBlockingStub(this.channel);
-    try {
-      stub.optimize(entity);
-      return true;
-    } catch (StatusRuntimeException e) {
-      e.printStackTrace();
-    }
-    return false;
-  }
-
-  public synchronized boolean dropEntityBlocking(Entity entity) {
-    final CottonDDLBlockingStub stub = CottonDDLGrpc.newBlockingStub(this.channel);
-    try {
-      stub.dropEntity(entity);
-      return true;
-    } catch (StatusRuntimeException e) {
-      if (e.getStatus().getCode() == Status.NOT_FOUND.getCode()) {
-        LOGGER.debug("entity {} was not dropped because it does not exist", entity.getName());
-      } else {
-        e.printStackTrace();
-      }
-    }
-    return false;
-  }
-
-  public synchronized ListenableFuture<CottontailGrpc.Status> createSchema(String schama) {
-    final CottonDDLFutureStub stub = CottonDDLGrpc.newFutureStub(this.channel);
-    return stub.createSchema(CottontailMessageBuilder.schema(schama));
-  }
-
-  public synchronized boolean createSchemaBlocking(String schema) {
-    ListenableFuture<CottontailGrpc.Status> future = this.createSchema(schema);
-    try {
-      future.get();
-      return true;
-    } catch (InterruptedException | ExecutionException e) {
-      LOGGER.error("error in createSchemaBlocking: {}", LogHelper.getStackTrace(e));
-      return false;
-    }
-  }
-
-  public synchronized void ensureSchemaBlocking(String schema) {
-    if (this.ensuredSchemas.contains(schema)) {
-      return;
-    }
-    final CottonDDLBlockingStub stub = CottonDDLGrpc.newBlockingStub(this.channel);
-    Iterator<Schema> existingSchemas = stub.listSchemas(Empty.getDefaultInstance());
-    boolean schemaExists = false;
-    while (existingSchemas.hasNext()) {
-      Schema existingSchema = existingSchemas.next();
-      if (existingSchema.getName().equals(schema)) {
-        schemaExists = true;
-      }
-    }
-    if (!schemaExists) {
-      this.createSchemaBlocking(schema);
-    }
-    this.ensuredSchemas.add(schema);
-  }
-
-  public boolean insert(List<InsertMessage> messages) {
 
-    final boolean[] status = {false, false}; /* {done, error}. */
-    final StreamObserver<CottontailGrpc.Status> observer = new StreamObserver<CottontailGrpc.Status>() {
+  public static final String CINEAST_SCHEMA = "cineast";
 
-      @Override
-      public void onNext(CottontailGrpc.Status value) {
-        LOGGER.trace("Tuple received: {}", value.getTimestamp());
-      }
+  /** The {@link ManagedChannel} used for communication. */
+  public final ManagedChannel channel;
 
-      @Override
-      public void onError(Throwable t) {
-        status[0] = true;
-        status[1] = true;
-        LOGGER.error("Error during insert. Everything was rolled back: {}", t.getMessage());
-      }
+  /** The {@link SimpleClient} instance that facilitates access to Cottontail DB. */
+  public final SimpleClient client;
```

Comment:
On `dev`, simply running `ping()` from Cineast on windows causes the following error Client-Side: `CANCELLED: RST_STREAM closed stream. HTTP/2 error code: CANCEL`

The cottontail-output is as follows: `Mar 15, 2021 9:32:59 AM io.grpc.internal.ServerCallImpl internalClose
WARNING: Cancelling the stream with status Status{code=INTERNAL, description=Completed without a response, cause=null}`.

I've fixed this in a1baab5e3ce2efcf758d7d7e2e6f9b46445edd34 on the cottontail `dev` branch